src/toast/map/pysm.py

```diff
@@ -62,37 +76,58 @@ def __init__(
     def init_sky(self, pysm_sky_config, pysm_precomputed_cmb_K_CMB):
         if pysm is None:
             raise RuntimeError("pysm not available")
-        initialized_sky_config = {}
         if pysm_precomputed_cmb_K_CMB is not None:
-            cmb = {
-                "model": "pre_computed",
-                "nside": self._nside,
-                "pixel_indices": self._local_pixels,
-            }
-            # PySM expects uK_CMB
-            cmb["A_I"], cmb["A_Q"], cmb["A_U"] = (
-                np.array(
-                    pysm.read_map(
-                        pysm_precomputed_cmb_K_CMB,
-                        self._nside,
-                        field=(0, 1, 2),
-                        pixel_indices=self._local_pixels,
-                        mpi_comm=self._comm,
-                    )
-                )
-                * 1e6
+            raise NotImplementedError(
+                "pysm_precomputed_cmb_K_CMB is not currently supported"
             )
-            initialized_sky_config["cmb"] = [cmb]
-            # remove cmb from the pysm string
-            pysm_sky_config.pop("cmb", None)
-        for name, model_id in pysm_sky_config.items():
-            initialized_sky_config[name] = pysm.nominal.models(
-                model_id, self._nside, self._local_pixels, mpi_comm=self._comm
-            )
-        return pysm.Sky(initialized_sky_config, mpi_comm=self._comm)
+            # cmb = {
+            #     "model": "pre_computed",
+            #     "nside": self._nside,
+            #     "pixel_indices": self._pixel_indices,
+            # }
+            # # PySM expects uK_CMB
+            # cmb["A_I"], cmb["A_Q"], cmb["A_U"] = (
+            #     np.array(
+            #         pysm.read_map(
+            #             pysm_precomputed_cmb_K_CMB,
```

Comment:
@keskitalo `pysm_precomputed_cmb_K_CMB` is a bit too specific, can we remove it?
it would be better to have an option in the pipeline to pass any custom PySM component with the new `pysm_component_objects` attribute of the PySM operator